### ISSUE TITLE
how to resolve the functionName with specialChar By the jeb FridaCodeGenerator.py

### ISSUE THREAD

**Issue by zxhubo** [smartdone/re_scripts#1]
Hi:
  Your FridaCodeGenerator.py is very userful for us.But I find a problem should be resolved.
  An app's code bas been obfuscated when build from code to an app. So if I open the app with Jeb,
this is show :
```
public static String ֏(aaa arg8, Object arg9, boolean arg10)
```
the functionName is special char ֏. When I use  FridaCodeGenerator.py to generate script，the error is :
```
'ascii' codec can't encode character u'\u058f' in position 0: ordinal not in range(128)
Place the cursor in the function you want to generate the Frida code, then run the script
```
I think should change this code'curent_addr ' where is:
```
if "(" in curent_addr:
                curent_addr = curent_addr.split("+")[0]
                print("current function: " + curent_addr.encode(encoding="utf-8"))
				
                m = FridaCodeGenerator.get_decompiled_method(self.dexunit, curent_addr)
```
But failed by try again and again,could you help me to resolve it ？

**Comment by zxhubo** [comment]
first:
you can add in this python file like this:
```
import sys
reload(sys)
sys.setdefaultencoding('utf8')
```

**Comment by zxhubo** [comment]
I got it!!! have been modified your FridaCodeGenerator.py to make it compatible between method_name is normally and special.

**Comment by zxhubo** [comment]
if a method iis a innerclass's method,this code :
```
def get_decompiled_method(dex, msig):
m = dex.getMethod(msig)
```
 then will return NoneType with m. so this code :+1: 
```
 m = FridaCodeGenerator.get_decompiled_method(self.dexunit, curent_addr)

method_name = m.getName()
```
the Exception is :'NoneType' object has no attribute 'getName'
how to resolve it ?

**Comment by zxhubo** [comment]
@smartdone

**Comment by smartdone** [comment]
Thanks for your feedback, I will see how to fix it.

**Comment by smartdone** [comment]
fixed on commit  275d21ea189cae5eac18a5f3b0c1875d4d0cf01b

**Comment by zxhubo** [comment]
@smartdone thx ,I know U
